src/components/cart/cart-item/cart-item.tsx

```diff
@@ -29,6 +29,10 @@ export const CartItem = ({
 
     const [quantity, setQuantity] = useState(item.quantity!);
 
+    useEffect(() => {
+        setQuantity(item.quantity!);
+    }, [item.quantity]);
```

Comment:
I'm afraid this can override the local `quantity`. For example, you update quanity from 1 to 2. While request is in progress you change it from 2 to 3. Then the request has finished and it overrides local state from 3 to 2. I think to prevent this you can do:
```js
if (!isUpdating) {
  setQuantity(item.quantity!);
}
```

Reply:
I'd rather disable changing quantity the same as checkout button on data updating. Because in your case we can loose data also (for e.g. if user changes quantity in Cart panel and cart page is open, it should update quantity on the cart-page as well)

Reply:
Disabling will not allow users to quickly change quantity by several units. They will have to do it slowly, one at a time.